packages/server/lib/controllers/connection/postConnection.ts

```diff
@@ -263,7 +281,24 @@ export const postPublicConnection = asyncWrapper<PostPublicConnection>(async (re
         return;
     }
 
+    let endUser: EndUser | undefined;
+    if (body.end_user) {
+        await db.knex.transaction(async (trx) => {
+            const endUserRes = await upsertEndUser(trx, { account, environment, endUserPayload: body.end_user });
+            if (endUserRes.isErr()) {
+                res.status(500).send({ error: { code: 'server_error', message: 'Failed to update end user' } });
+                return;
+            }
+
+            endUser = endUserRes.value;
+
+            await linkConnection(trx, { endUserId: endUserRes.value.id, connection: updatedConnection.connection });
```

Comment:
should the Result be checked and return a 500 if it fails?

Reply:
this one is throwing